File: mobile/android/geckoview/src/main/java/org/mozilla/geckoview/WebNotificationDelegate.java
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.geckoview;

import androidx.annotation.AnyThread;
import androidx.annotation.NonNull;
import org.mozilla.gecko.annotation.WrapForJNI;

public interface WebNotificationDelegate {
  /**
   * This is called when a new notification is created.
   *
   * @param notification The WebNotification received.
   */
  @AnyThread
  @WrapForJNI
  default void onShowNotification(@NonNull final WebNotification notification) {}

  /**
   * This is called when an existing notification is closed.
   *
   * @param notification The WebNotification received.
   */
  @AnyThread
  @WrapForJNI
  default void onCloseNotification(@NonNull final WebNotification notification) {}
}

File: mobile/android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/webnotifications/WebNotificationDelegate.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.components.concept.engine.webnotifications

/**
 * Notifies applications or other components of engine events related to web
 * notifications e.g. an notification is to be shown or is to be closed
 */
interface WebNotificationDelegate {
    /**
     * Invoked when a web notification is to be shown.
     *
     * @param webNotification The web notification intended to be shown.
     */
    fun onShowNotification(webNotification: WebNotification) = Unit

    /**
     * Invoked when a web notification is to be closed.
     *
     * @param webNotification The web notification intended to be closed.
     */
    fun onCloseNotification(webNotification: WebNotification) = Unit
}
